--- linux-3.9.7-r1/arch/mips/brcmstb/Kconfig	2014-02-26 13:34:47.000000000 +0900
+++ linux-3.9.7-r1.new/arch/mips/brcmstb/Kconfig	2014-03-13 21:55:59.597462948 +0900
@@ -164,7 +164,6 @@
 	select BRCM_HAS_PCI23
 	select BRCM_HAS_EMAC_0
 	select BRCM_HAS_EMAC_1
-	select BRCM_HAS_NOR
 	select BRCM_HAS_NAND_OLD_MAJOR_1
 	select BRCM_HAS_NAND_MINOR_1
 	select BRCM_HAS_NAND_MINOR_2
@@ -1081,6 +1080,7 @@
 
 config BRCM_HAS_NOR
 	bool
+	default n
 
 config BRCM_HAS_NAND_MINOR_0
 	bool
