diff --git a/plugin/controllers/ajax.py b/plugin/controllers/ajax.py
index 0b16ddd..fc179c8 100644
--- a/plugin/controllers/ajax.py
+++ b/plugin/controllers/ajax.py
@@ -87,6 +87,16 @@ class AjaxController(BaseController):
 			model = "et5x00"
 		elif model == "et4000" :
 			model = "et4x00"
+		elif model == "mediabox":
+			model = "Mediabox HD LX-1"
+		elif model == "optimussos1":
+			model = "Optimuss OS1"
+		elif model == "optimussos2":
+			model = "Optimuss OS2"
+		elif model == "optimussos1plus":
+			model = "Optimuss OS1+"
+		elif model == "optimussos2plus":
+			model = "Optimuss OS2+"
 		elif model == "xp1000" :
 			model = "xp1000"		
 		if fileExists(getPublicPath("/images/boxes/" + model + ".jpg")):
@@ -106,8 +116,8 @@ class AjaxController(BaseController):
 	def P_screenshot(self, request):
 		box = {}
 		box['brand'] = "dmm"
-		if fileExists("/proc/stb/info/vumodel"):
-			box['brand'] = "vuplus"
+		if fileExists("/proc/stb/info/hwmodel"):
+			box['brand'] = "4D"
 		elif fileExists("/proc/stb/info/azmodel"):
 			box['brand'] = "azbox"
 		elif fileExists("/proc/stb/info/gbmodel"):
diff --git a/plugin/controllers/base.py b/plugin/controllers/base.py
index 89577d0..5fc0817 100755
--- a/plugin/controllers/base.py
+++ b/plugin/controllers/base.py
@@ -145,10 +145,10 @@ class BaseController(resource.Resource):
 			ret['box'] = "gigablue"
 		if fileExists("/proc/stb/info/boxtype"):
 			ret['box'] = open("/proc/stb/info/boxtype").read().strip().lower()
-		elif fileExists("/proc/stb/info/vumodel"):
-			ret['box'] = open("/proc/stb/info/vumodel").read().strip().lower()
 		elif fileExists("/proc/stb/info/azmodel"):
 			ret['box'] = open("/proc/stb/info/model").read().strip().lower()
+		elif fileExists("/proc/stb/info/hwmodel"):
+			ret['box'] = open("/proc/stb/info/hwmodel").read().strip().lower()
 			
 		if ret["box"] in ("solo", "duo", "uno", "solo2", "duo2"):
 			ret["remote"] = "vu_normal"
@@ -178,6 +178,14 @@ class BaseController(resource.Resource):
 			ret["remote"] = "ini-7000"
 		elif ret["box"] == "xp1000":
 			ret["remote"] = "xp1000"
+		elif ret["box"] in ( "tmtwinoe", "tm2toe", "tmsingleoe", "tmnanooe" ,"tmnanosuper" ):
+			ret["remote"] = "tm_models"
+		elif ret["box"] in ( "ios100", "ios200", "ios300", "force1", "force1plus", "force2" ):
+			ret["remote"] = "iqon_models"
+		elif ret["box"] in ( "mediabox" ):
+			ret["remote"] = "mediabox"
+		elif ret["box"] in ( "optimussos1", "optimussos2", "optimussos1plus", "optimussos2plus" ):
+			ret["remote"] = "edition"
 		else:
 			ret["remote"] = "dmm"
 		extras = []
diff --git a/plugin/controllers/models/info.py b/plugin/controllers/models/info.py
index 8fbfa6e..78424fd 100644
--- a/plugin/controllers/models/info.py
+++ b/plugin/controllers/models/info.py
@@ -87,11 +87,11 @@ def getInfo():
 		elif model.startswith("xp"):
 		    brand = "MaxDigital"
  		f.close()
-	elif fileExists("/proc/stb/info/vumodel"):
-		brand = "VuPlus"
-		f = open("/proc/stb/info/vumodel",'r')
- 		model = f.readline().strip().lower()
- 		f.close()
+	elif fileExists("/proc/stb/info/hwmodel"):
+		brand = "4D"
+		f = open("/proc/stb/info/hwmodel",'r')
+		model = f.readline().strip()
+		f.close()
 	elif fileExists("/proc/stb/info/azmodel"):
 		brand = "AZBox"
 		f = open("/proc/stb/info/model",'r')
