diff --git a/plugin/controllers/ajax.py b/plugin/controllers/ajax.py
index 0b16ddd..f55a401
--- a/plugin/controllers/ajax.py
+++ b/plugin/controllers/ajax.py
@@ -108,6 +108,8 @@ class AjaxController(BaseController):
 		box['brand'] = "dmm"
 		if fileExists("/proc/stb/info/vumodel"):
 			box['brand'] = "vuplus"
+		elif fileExists("/proc/stb/info/hwmodel"):
+			box['brand'] = '"4D"
 		elif fileExists("/proc/stb/info/azmodel"):
 			box['brand'] = "azbox"
 		elif fileExists("/proc/stb/info/gbmodel"):
diff --git a/plugin/controllers/base.py b/plugin/controllers/base.py
index 89577d0..4953cf2 100755
--- a/plugin/controllers/base.py
+++ b/plugin/controllers/base.py
@@ -149,6 +149,8 @@ class BaseController(resource.Resource):
 			ret['box'] = open("/proc/stb/info/vumodel").read().strip().lower()
 		elif fileExists("/proc/stb/info/azmodel"):
 			ret['box'] = open("/proc/stb/info/model").read().strip().lower()
+		elif fileExists("/proc/stb/info/hwmodel"):
+			ret['box'] = open("/proc/stb/info/hwmodel").read().strip().lower()
 			
 		if ret["box"] in ("solo", "duo", "uno", "solo2", "duo2"):
 			ret["remote"] = "vu_normal"
@@ -178,6 +180,14 @@ class BaseController(resource.Resource):
 			ret["remote"] = "ini-7000"
 		elif ret["box"] == "xp1000":
 			ret["remote"] = "xp1000"
+		elif ret["box"] in ( "tmtwinoe", "tm2toe", "tmsingleoe", "tmnanooe" )
+			ret["remote"] = "tm_models"
+		elif ret["box"] in ( "ios100", "ios200", "ios300", "force1", "force1plus", "force2", "tmnano2tsuper" )
+			ret["remote"] = "iqon_models"
+		elif ret["box"] in ( "mediabox" )
+			ret["remote"] = "mediabox"
+		elif ret["box"] in ( "optimussos1", "optimussos2", "optimussos1plus", "optimussos2plus" )
+			ret["remote"] = "edition"
 		else:
 			ret["remote"] = "dmm"
 		extras = []
--- a/plugin/controllers/models/info.py
+++ b/plugin/controllers/models/info.py
@@ -92,6 +92,11 @@ def getInfo():
 		f = open("/proc/stb/info/vumodel",'r')
  		model = f.readline().strip().lower()
  		f.close()
+	elif fileExists("/proc/stb/info/hwmodel"):
+		brand = "4D"
+		f = open("/proc/stb/info/hwmodel",'r')
+		model = f.readline().strip()
+		f.close()
 	elif fileExists("/proc/stb/info/azmodel"):
 		brand = "AZBox"
 		f = open("/proc/stb/info/model",'r')
